src/toolchain/export-mac-and-ios.md

```diff
@@ -0,0 +1,316 @@
+We will create corresponding environment variables and use a script to sign, so it's easier to run. Here are the environment variables needed:
+
+- `APPLE_CERT_BASE64`
+- `APPLE_CERT_PASSWORD`
+- `APPLE_DEV_ID`
+- `APPLE_DEV_TEAM_ID`
+- `APPLE_DEV_PASSWORD`
+- `APPLE_DEV_APP_ID`
+
+Firstly, make sure to enroll your Apple ID to the Developer Program:
+
+- Create an Apple ID if you don't have one already.
+- Use your Apple ID to register in the Apple Developer Program by going to [developer.apple.com](https://developer.apple.com).
+- Accept all agreements from the Apple Developer Page.
+
+
+### `APPLE_DEV_ID` - Apple ID
+
+
+Your email used for your Apple ID.
+
+```sh
+APPLE_DEV_ID = email@provider.com
+```
+
+
+### APPLE_DEV_TEAM_ID - Apple Team ID
+
+
+Go to [developer.apple.com](https://developer.apple.com). Go to account.
+
+Go to membership details. Copy Team ID.
+
+```sh
+APPLE_DEV_TEAM_ID = 1ABCD23EFG
+```
+
+
+### APPLE_DEV_PASSWORD - Apple App-Specific Password
+
+
+Create Apple App-Specific Password. Copy the password.
+
+```sh
+APPLE_DEV_PASSWORD = abcd-abcd-abcd-abcd
+```
+
+
+### `APPLE_CERT_BASE64`, `APPLE_CERT_PASSWORD` and `APPLE_DEV_APP_ID`
+
+
+Go to [developer.apple.com](https://developer.apple.com). Go to account.
+
+Go to certificates.
+
+Click on + at Certificates tab. Create Developer ID Application. Click Continue.
+
+Leave profile type as is. Create a certificate signing request from a Mac. You can use your own name and email address. Save the file to disk.
+You will get a file called `CertificateSigningRequest.certSigningRequest`. Upload it to the Developer ID Application request. Click Continue.
+
+Download the certificate. You will get a file `developerID_application.cer`.
+
+On a Mac, right click and select open. Add it to the login keychain.
+In the Keychain Access app that opened, log into Keychain tab, go to Keys, sort by date modified,
+and expand your key (the key should have the name you entered at _Common Name_).
+Right click the expanded certificate, get info, and copy the text at _Details -> Subject Name -> Common Name_. For example:
+
+```sh
+APPLE_DEV_APP_ID = Developer ID Application: Common Name (1ABCD23EFG)
+```
+
+Then, select the certificate, right click and click export. At file format select `p12`. When exporting, set a password for the certificate.
+This will be the value of `APPLE_CERT_PASSWORD`. You will get a `Certificates.p12` file.
+
+For example:
+
+```sh
+APPLE_CERT_PASSWORD = <password_set_when_exporting_p12>
+```
+
+Then you need to make a base64 file out of it, by running:
+
+```sh
+base64 -i Certificates.p12 -o Certificates.base64
+```
+
+Copy the contents of the generated file, e.g.:
+APPLE_CERT_BASE64 = ...(A long text file)
+
+After these secrets are obtained, all that remains is to set them as environment variables.
+Afterwards you can use the following script
+for signing [ci-sign-macos.ps1](https://github.com/appsinacup/godot-rapier-physics/blob/main/scripts/ci-sign-macos.ps1).
```

Comment:
Running random scripts from a third-party repository is both a bit insecure and brittle. It will break as soon as someone changes file name/path, and people can introduce code at any time.

The fact that the script has *zero* comments and is itself a fork of https://github.com/godot-jolt/godot-jolt/blob/master/scripts/ci_sign_macos.ps1 doesn't make it more trustworthy, either.

Is this the easiest/best way? Are there official resources that it can be complemented with?

Reply:
There is a tool, fastlane, which can be used. However I haven't used it, and for my use cases I needed a script as I wanted it automated.

I would propose addint it inline then, and we can give credits to the original script.

Reply:
What does the godot-rapier-physics version change compared to the godot-jolt one?

Reply:
I guess I was not owner of that repo, so thats why I didnt put that, as I tested this version of script and I would know what happens to it. I also put some of the logic in the godot-cpp-template repo and credited godot-jolt repo.

Could do same here, copy paste contents of script? It's basically some steps that are tedious to do by hand. Most people however if they do this, they would not do this step, if they build from godot(eg. godot automated this signing step). So this is only for people who export the lib as is, and those people might anyway use github actions for that. For that use case I could just give a reusable external action, eg. https://github.com/godotengine/godot-cpp-template/blob/main/.github/actions/sign/action.yml

Reply:
I could describe the steps individually and not use the script in this tutorial, and credit the creator somewhere.

Reply:
I think it's OK to link to an external script if it's somewhat reputed (like godot-jolt). We should just add a disclaimer that the user is responsible for the security and up-to-dateness.

As for describing the script's step, I'm not sure if that doesn't widen the scope of the tutorial too much. Given that the script is already quite long, explaining it would be its own huge page in the book, no?

Reply:
Ok, looking at the jolt one, looks it has less flag's on some commands, but probably works just as well. My only fear if linking to that and if it doesn't work, would be hard to change that.

That being said, updating with that script, if anything it's a good starting point for people willing do signing by hand (that step is optional if you build a game for eg, it's only needed in case you want to export the lib as is)